### ISSUE TITLE
Difficult to move focus from DateInput input to its popover

### ISSUE THREAD

**Issue by adidahiya** [palantir/blueprint#4812]
#### Environment

- __Package version(s)__: datetime v3.23.6
- __Operating System__: macOS
- __Browser name and version__: Chrome

#### Code Sandbox

Link to a minimal repro: https://codesandbox.io/s/blueprint-sandbox-forked-6wt6o?file=/src/DatetimeExample.tsx

#### Steps to reproduce

1. Focus on input
1. Press TAB key

#### Actual behavior

Focus briefly moves to the calendar inside the popover, but then the popover closes

![2021-07-20 16 01 32](https://user-images.githubusercontent.com/723999/126387988-07a0b230-8265-4389-907b-50a9a635b4bb.gif)

#### Expected behavior

Users should have the affordance to navigate to the calendar inside the popover to select a date by using only the keyboard.

#### Possible solution

This is unlikely to be fixed simply by moving to Popover2, as it's still a problem in the [v4 pre-release docs](https://blueprintjs.com/docs/versions/4/#datetime/dateinput).

**Comment by adidahiya** [comment]
This was actually fixed for DateInput, but that fix did not make it to DateInput2.

DateInput (@blueprintjs/datetime 4.4.35)

![2023-06-05 11 07 04](https://github.com/palantir/blueprint/assets/723999/d305f257-6346-4755-b0d4-8802620899e7)

DateInput2 (@blueprintjs/datetime2 0.9.34)

![2023-06-05 11 07 31](https://github.com/palantir/blueprint/assets/723999/234ba48a-7120-4af5-9ff3-3b982ae2a47c)

**Comment by ericjeney** [comment]
Closing since this now works in `DateInput3`.
